rialto_airflow/publish/publication.py

```diff
@@ -484,3 +496,207 @@ def _citation_count(row) -> str | int | None:
     # drop any string or None values
     counts = [count for count in counts if isinstance(count, int)]
     return max(counts) if counts else None
+
+
+def _author_list_names(row) -> list[Any]:
+    """
+    Get a pipe delimited list of all the author names.
+    """
+    names = first(
+        row,
+        rules=[
+            JsonPathRule(
+                "openalex_json", "authorships[*].author.display_name", return_list=True
+            ),
+            FuncRule("dim_json", _dim_author_list_names),
+            FuncRule("pubmed_json", _pubmed_author_list_names),
+            JsonPathRule(
+                "wos_json",
+                "static_data.summary.names.name[*].display_name",
+                return_list=True,
+            ),
+            JsonPathRule(
+                "wos_json",
+                "static_data.summary.names.name.display_name",
+                return_list=True,
+            ),
```

Comment:
there are multiple wos_json rules here because sometimes we get a single object rather than a list